qutebrowser/browser/hints.py

```diff
@@ -767,6 +781,48 @@ def current_mode(self):
 
         return self._context.hint_mode
 
+    def _get_visible_hints(self):
+        """Get elements which are currently visible."""
+        visible = {}
+        for string, elem in self._context.elems.items():
+            try:
+                if not self._is_hidden(elem.label):
+                    visible[string] = elem
+            except webelem.Error:
+                pass
+        return visible
+
+    def _handle_auto_follow(self, keystr="", filterstr="", visible=None):
+        """Handle the auto-follow option."""
+        if visible is None:
+            visible = self._get_visible_hints()
+
+        if len(visible) != 1:
+            return
+
+        auto_follow = config.get('hints', 'auto-follow')
+
+        if auto_follow == "always":
+            follow = True
+        elif auto_follow == "unique-match":
+            follow = (keystr or filterstr)
+        elif auto_follow == "full-match":
+            elemstr = str(list(visible.values())[0].elem)
+            filter_match = self._filter_matches_exactly(filterstr, elemstr)
+            follow = keystr in visible or filter_match
```

Comment:
Since `in` is also just an operator, I'm not 100% sure if this does the right thing - I'd be more comfortable with `(keystr in visible) or filter_match` :wink: